programs/buck_version.py

```diff
@@ -80,17 +80,13 @@ def get_clean_buck_version(dirpath, allow_dirty=False):
 
 
 def get_dirty_buck_version(dirpath):
-    git_tree_in = check_output(
-        ['git', 'log', '-n1', '--pretty=format:%T', 'HEAD', '--'],
-        cwd=dirpath).strip()
-
     with EmptyTempFile(prefix='buck-git-index') as index_file:
+        subprocess.check_call(
+            ['git', 'read-tree', '--index-output', index_file.name, '--reset', 'HEAD'],
```

Comment:
I realize this doesn't change the checked out files, but should we pass `--dry-run` here as well?  Not 100% clear to me from the man page.